src/VisualStudio/IntegrationTest/TestUtilities/Helper.cs

```diff
@@ -66,7 +66,7 @@ public static T Retry<T>(Func<T> action, TimeSpan delay, TimeSpan timeout)
                     System.Threading.Thread.Sleep(delay);
                 }
             }
-            while (beginTime.Add(timeout) > DateTime.UtcNow);
+            while (timeout == TimeSpan.MaxValue || beginTime.Add(timeout) > DateTime.UtcNow);
```

Comment:
I thought, in the initial design, we had decided to not allow any timeouts and to just retry indefinitely....

@jasonmalinowski, @DustinCampbell